lib/web/assistant.go

```diff
@@ -358,17 +358,36 @@ func (h *Handler) reportTokenUsage(usedTokens *tokens.TokenCount, conversationID
 	}
 	h.assistantLimiter.ReserveN(time.Now(), extraTokens)
 
-	usageEventReq := &proto.SubmitUsageEventRequest{
-		Event: &usageeventsv1.UsageEventOneOf{
-			Event: &usageeventsv1.UsageEventOneOf_AssistCompletion{
-				AssistCompletion: &usageeventsv1.AssistCompletionEvent{
-					ConversationId:   conversationID,
-					TotalTokens:      int64(promptTokens + completionTokens),
-					PromptTokens:     int64(promptTokens),
-					CompletionTokens: int64(completionTokens),
+	var usageEventReq *proto.SubmitUsageEventRequest
+	switch {
+	case conversationID != "":
+		usageEventReq = &proto.SubmitUsageEventRequest{
+			Event: &usageeventsv1.UsageEventOneOf{
+				Event: &usageeventsv1.UsageEventOneOf_AssistCompletion{
+					AssistCompletion: &usageeventsv1.AssistCompletionEvent{
+						ConversationId:   conversationID,
+						TotalTokens:      int64(promptTokens + completionTokens),
+						PromptTokens:     int64(promptTokens),
+						CompletionTokens: int64(completionTokens),
+					},
 				},
 			},
-		},
+		}
+	case action != "":
+		usageEventReq = &proto.SubmitUsageEventRequest{
+			Event: &usageeventsv1.UsageEventOneOf{
+				Event: &usageeventsv1.UsageEventOneOf_AssistAction{
+					AssistAction: &usageeventsv1.AssistAction{
+						Action:           action,
+						TotalTokens:      int64(promptTokens + completionTokens),
+						PromptTokens:     int64(promptTokens),
+						CompletionTokens: int64(completionTokens),
+					},
+				},
+			},
+		}
+	default:
+		h.log.Errorf("unknown mix supplied: '%s', '%s'", conversationID, action)
```

Comment:
The only way this would happen is if both are unset, so it will print `'', ''`.

Additionally, `unknown mix supplied` might be pretty confusing to diagnose. Can we improve the error message here? (and maybe make it a warning instead?)

Reply:
good catch, fixed. Thanks